### DISCUSSION TITLE
Spark user info from the agent

### DISCUSSION THREAD

**Discussion by vishalag001** [AbsaOSS/spline-spark-agent#473] [General]
Currently, the spark agent is not capturing the sparkUser who has initiated the job. It is readily available in the sparkContext. The user info can go into the extras of executionPlan or the event.

Benefits of User info:

- Help enterprises to identify individual teams and get back with any findings regarding lineage.
- A good metadata attribute from a lineage perspective.

I am able to capture sparkUser info in local. Happy to add it to open source.

@wajda Any thoughts on this ? Happy to brainstorm

**Comment by wajda** [comment]
Have you tried `MetadataCollectingFilter`? This is a post-processing lineage filter that is specifically designed for capturing a user-specific extra information, like the context related info, environment variables, user identity etc etc.
It's available in the development branch (ver 1.0.0-SNAPSHOT). It's not properly documented yet, but you can see example of usage in the [test case ](https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/test/scala/za/co/absa/spline/harvester/postprocessing/extra/MetadataCollectingFilterSpec.scala) or the [example Spark app](https://github.com/AbsaOSS/spline-spark-agent/blob/develop/examples/src/main/scala/za/co/absa/spline/SparkApp.scala)

So your capturing rules would look something like this:
```properties
spline.postProcessingFilter.userExtraMeta.rules=               \
{                                                              \
  "executionEvent": {                                          \
    "sparkUser": { "$js": "session.sparkContext.sparkUser" }   \
  }                                                            \
}
```
Don't forget to enable the filter with something like this:
```properties
spline.postProcessingFilter=userExtraMeta
```

**Comment by wajda** [comment]
I've created a related ticket #474

**Reply by vishalag001** [reply]
https://github.com/AbsaOSS/spline-spark-agent/pull/475
Raised a PR
